File: discovery-engine/pocs/github_osint.py
import sys
import requests

def run_github_osint(startup_name):
    url = f"https://api.github.com/search/users?q=company:{startup_name}"
    headers = {"Accept": "application/vnd.github.v3+json"}
    print(f"Querying GitHub API: {url}")
    
    try:
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        data = response.json()
        
        items = data.get('items', [])
        if not items:
            print(f"No users found associated with company: {startup_name}")
            return
            
        print(f"Found {len(items)} users. Fetching details for up to 5...\n")
        
        for i, item in enumerate(items[:5]):
            user_url = item.get('url')
            if user_url:
                user_resp = requests.get(user_url, headers=headers)
                if user_resp.status_code == 200:
                    user_data = user_resp.json()
                    name = user_data.get('name') or item.get('login')
                    company = user_data.get('company', 'Unknown')
                    bio = user_data.get('bio', '')
                    blog = user_data.get('blog', '')
                    
                    print(f"[{i+1}] Name/Login: {name}")
                    print(f"    Company: {company}")
                    print(f"    Bio: {bio}")
                    print(f"    Blog/Website: {blog}")
                    print(f"    GitHub Profile: {item.get('html_url')}\n")
    except Exception as e:
        print(f"Error during GitHub API request: {e}")

if __name__ == "__main__":
    target = "Zenskar"
    if len(sys.argv) > 1:
        target = sys.argv[1]
    run_github_osint(target)

File: discovery-engine/pocs/github_osint.js
const fs = require('fs');
const path = require('path');

const COMPANY = process.argv[2] || "Cashify";
const OUTPUT_FILE = path.join(__dirname, 'output_gh.txt');

async function main() {
    try {
        console.log(`Searching GitHub for employees of ${COMPANY}...`);
        
        const response = await fetch(`https://api.github.com/search/users?q=${COMPANY}+type:user`, {
            headers: {
                'User-Agent': 'Node.js-OSINT-Script'
            }
        });

        if (!response.ok) {
            throw new Error(`GitHub API error: ${response.status} ${response.statusText}`);
        }

        const data = await response.json();
        const users = data.items || [];
        
        console.log(`Found ${users.length} users. Fetching details...`);
        
        let outputText = `GitHub Employees for ${COMPANY}:\n=================================\n\n`;

        for (const user of users) {
            console.log(`Fetching details for ${user.login}...`);
            const userResponse = await fetch(user.url, {
                headers: {
                    'User-Agent': 'Node.js-OSINT-Script'
                }
            });

            if (userResponse.ok) {
                const userData = await userResponse.json();
                outputText += `Name: ${userData.name || 'N/A'}\n`;
                outputText += `Login: ${userData.login}\n`;
                outputText += `Company: ${userData.company || 'N/A'}\n`;
                outputText += `Location: ${userData.location || 'N/A'}\n`;
                outputText += `Email: ${userData.email || 'N/A'}\n`;
                outputText += `Blog/Website: ${userData.blog || 'N/A'}\n`;
                outputText += `Bio: ${userData.bio || 'N/A'}\n`;
                outputText += `GitHub URL: ${userData.html_url}\n`;
                outputText += `---------------------------------\n`;
            } else {
                console.warn(`Failed to fetch details for ${user.login}: ${userResponse.status}`);
            }
            // Add a small delay to avoid rate limits
            await new Promise(resolve => setTimeout(resolve, 500));
        }

        fs.writeFileSync(OUTPUT_FILE, outputText);
        console.log(`Successfully wrote data to ${OUTPUT_FILE}`);
        
    } catch (error) {
        console.error("Error occurred:", error.message);
        process.exit(1);
    }
}

main();
